### PR TITLE
JAMES-2855 Dependency cleanup

### PR DESCRIPTION
**Pull request by chibenwa**
Using `mvn dependency:analyze` try to remove as much dependencies as we can!

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] [ERROR] Errors: 
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] [ERROR]   JamesAppSpringMainTest.testServer:27 » BeanDefinitionStore Unexpected exceptio...
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] [ERROR] org.apache.james.app.spring.JamesSpringContextTest.springShouldLoadAndAddOnlyOneQuotaUpdaterListener(org.apache.james.app.spring.JamesSpringContextTest)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] [ERROR]   Run 1: JamesSpringContextTest.setup:45 » BeanDefinitionStore Unexpected exception par...
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] [ERROR]   Run 2: JamesSpringContextTest.tearDown:52 NullPointer
```

**Comment by chibenwa** [comment]
```
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] java.net.MalformedURLException: unknown protocol: classpath
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.base/java.net.URL.<init>(URL.java:634)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.base/java.net.URL.<init>(URL.java:523)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.base/java.net.URL.<init>(URL.java:470)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at org.springframework.beans.factory.xml.ResourceEntityResolver.resolveEntity(ResourceEntityResolver.java:79)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.xml/com.sun.org.apache.xerces.internal.util.EntityResolverWrapper.resolveEntity(EntityResolverWrapper.java:111)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLEntityManager.resolveEntity(XMLEntityManager.java:1136)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.xml/com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaLoader.resolveDocument(XMLSchemaLoader.java:662)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.xml/com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.findSchemaGrammar(XMLSchemaValidator.java:2694)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.xml/com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.handleStartElement(XMLSchemaValidator.java:2069)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.xml/com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.startElement(XMLSchemaValidator.java:829)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:374)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl$NSContentDriver.scanRootElementHook(XMLNSDocumentScannerImpl.java:613)
[3d92ad8e0f8ae9e67d886d7b04ba0764563031ec] 	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:3063)
```

**Comment by chibenwa** [comment]
`[203/224]`

**Comment by Arsnael** [comment]
```
[2a188129567cd20457537f032b010b03496a2af3] [ERROR] Tests run: 7, Failures: 0, Errors: 7, Skipped: 0, Time elapsed: 2.142 s <<< FAILURE! - in org.apache.james.JPAJamesServerWithAuthenticatedDatabaseSqlValidationTest
[2a188129567cd20457537f032b010b03496a2af3] [ERROR] connectLMTPServerShouldSendShabangOnConnect{GuiceJamesServer}  Time elapsed: 0.425 s  <<< ERROR!
[2a188129567cd20457537f032b010b03496a2af3] org.apache.commons.configuration2.ex.ConfigurationException: Unable to init mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: javax.mail.MessagingException: Can not load mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: java.lang.ClassNotFoundException: Bounce
[2a188129567cd20457537f032b010b03496a2af3] 
[2a188129567cd20457537f032b010b03496a2af3] [ERROR] hostnameShouldBeUsedAsDefaultDomain{GuiceJamesServer}  Time elapsed: 0.326 s  <<< ERROR!
[2a188129567cd20457537f032b010b03496a2af3] org.apache.commons.configuration2.ex.ConfigurationException: Unable to init mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: javax.mail.MessagingException: Can not load mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: java.lang.ClassNotFoundException: Bounce
[2a188129567cd20457537f032b010b03496a2af3] 
[2a188129567cd20457537f032b010b03496a2af3] [ERROR] connectIMAPServerShouldSendShabangOnConnect{GuiceJamesServer}  Time elapsed: 0.316 s  <<< ERROR!
[2a188129567cd20457537f032b010b03496a2af3] org.apache.commons.configuration2.ex.ConfigurationException: Unable to init mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: javax.mail.MessagingException: Can not load mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: java.lang.ClassNotFoundException: Bounce
[2a188129567cd20457537f032b010b03496a2af3] 
[2a188129567cd20457537f032b010b03496a2af3] [ERROR] connectSMTPServerShouldSendShabangOnConnect{GuiceJamesServer}  Time elapsed: 0.25 s  <<< ERROR!
[2a188129567cd20457537f032b010b03496a2af3] org.apache.commons.configuration2.ex.ConfigurationException: Unable to init mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: javax.mail.MessagingException: Can not load mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: java.lang.ClassNotFoundException: Bounce
[2a188129567cd20457537f032b010b03496a2af3] 
[2a188129567cd20457537f032b010b03496a2af3] [ERROR] hostnameShouldBeRetrievedWhenRestarting{GuiceJamesServer}  Time elapsed: 0.291 s  <<< ERROR!
[2a188129567cd20457537f032b010b03496a2af3] org.apache.commons.configuration2.ex.ConfigurationException: Unable to init mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: javax.mail.MessagingException: Can not load mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: java.lang.ClassNotFoundException: Bounce
[2a188129567cd20457537f032b010b03496a2af3] 
[2a188129567cd20457537f032b010b03496a2af3] [ERROR] connectOnSecondaryIMAPServerIMAPServerShouldSendShabangOnConnect{GuiceJamesServer}  Time elapsed: 0.23 s  <<< ERROR!
[2a188129567cd20457537f032b010b03496a2af3] org.apache.commons.configuration2.ex.ConfigurationException: Unable to init mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: javax.mail.MessagingException: Can not load mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: java.lang.ClassNotFoundException: Bounce
[2a188129567cd20457537f032b010b03496a2af3] 
[2a188129567cd20457537f032b010b03496a2af3] [ERROR] connectPOP3ServerShouldSendShabangOnConnect{GuiceJamesServer}  Time elapsed: 0.303 s  <<< ERROR!
[2a188129567cd20457537f032b010b03496a2af3] org.apache.commons.configuration2.ex.ConfigurationException: Unable to init mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: javax.mail.MessagingException: Can not load mailet Bounce
[2a188129567cd20457537f032b010b03496a2af3] Caused by: java.lang.ClassNotFoundException: Bounce
[2a188129567cd20457537f032b010b03496a2af3] 
[2a188129567cd20457537f032b010b03496a2af3] [INFO] 
[2a188129567cd20457537f032b010b03496a2af3] [INFO] Results:
[2a188129567cd20457537f032b010b03496a2af3] [INFO] 
[2a188129567cd20457537f032b010b03496a2af3] [ERROR] Errors: 
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerTest.connectIMAPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerTest.connectLMTPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerTest.connectOnSecondaryIMAPServerIMAPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerTest.connectPOP3ServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerTest.connectSMTPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerTest.hostnameShouldBeRetrievedWhenRestarting » Configuration Una...
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerTest.hostnameShouldBeUsedAsDefaultDomain » Configuration Unable ...
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithAuthenticatedDatabaseSqlValidationTest.connectIMAPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithAuthenticatedDatabaseSqlValidationTest.connectLMTPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithAuthenticatedDatabaseSqlValidationTest.connectOnSecondaryIMAPServerIMAPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithAuthenticatedDatabaseSqlValidationTest.connectPOP3ServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithAuthenticatedDatabaseSqlValidationTest.connectSMTPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithAuthenticatedDatabaseSqlValidationTest.hostnameShouldBeRetrievedWhenRestarting » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithAuthenticatedDatabaseSqlValidationTest.hostnameShouldBeUsedAsDefaultDomain » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithNoDatabaseAuthenticaticationSqlValidationTest.connectIMAPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithNoDatabaseAuthenticaticationSqlValidationTest.connectLMTPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithNoDatabaseAuthenticaticationSqlValidationTest.connectOnSecondaryIMAPServerIMAPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithNoDatabaseAuthenticaticationSqlValidationTest.connectPOP3ServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithNoDatabaseAuthenticaticationSqlValidationTest.connectSMTPServerShouldSendShabangOnConnect » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithNoDatabaseAuthenticaticationSqlValidationTest.hostnameShouldBeRetrievedWhenRestarting » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JPAJamesServerWithNoDatabaseAuthenticaticationSqlValidationTest.hostnameShouldBeUsedAsDefaultDomain » Configuration
[2a188129567cd20457537f032b010b03496a2af3] [ERROR]   JamesCapabilitiesServerTest.startShouldSucceedWhenRequiredCapabilities » Configuration
```
I'm a bit on a loss regarding that one...

**Comment by Arsnael** [comment]
```
[3e006480051a42fdc84b8f9d5f3178b117190c1f] 10:52:18.234 [1;31m[ERROR][0;39m o.a.j.m.l.AbstractStateMailetProcessor - Unable to init mailet MetricsMailet
[3e006480051a42fdc84b8f9d5f3178b117190c1f] java.lang.ClassNotFoundException: MetricsMailet
[3e006480051a42fdc84b8f9d5f3178b117190c1f] 	at org.apache.james.utils.GuiceGenericLoader.lambda$locateClass$3(GuiceGenericLoader.java:50)
[3e006480051a42fdc84b8f9d5f3178b117190c1f] 	at java.util.Optional.orElseThrow(Optional.java:290)
[3e006480051a42fdc84b8f9d5f3178b117190c1f] 	at org.apache.james.utils.GuiceGenericLoader.locateClass(GuiceGenericLoader.java:50)
[3e006480051a42fdc84b8f9d5f3178b117190c1f] 	at org.apache.james.utils.GuiceGenericLoader.instanciate(GuiceGenericLoader.java:41)
[3e006480051a42fdc84b8f9d5f3178b117190c1f] 	at org.apache.james.utils.GuiceMailetLoader.getMailet(GuiceMailetLoader.java:55)
[3e006480051a42fdc84b8f9d5f3178b117190c1f] 	... 31 common frames omitted
```
tests are :green_apple: , but looks like something is still  missing

**Comment by chibenwa** [comment]
`RabbitMQEventBusTest$ConcurrentTest.concurrentDispatchShouldDeliverAllEventsToListenersWithSingleEventBus » AlreadyClosed`

test this please

File: mpt/impl/imap-mailbox/cyrus/pom.xml [hunk 1]
```diff
@@ -39,6 +39,10 @@
             <groupId>org.awaitility</groupId>
             <artifactId>awaitility</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.google.inject</groupId>
+            <artifactId>guice</artifactId>
```
Comment (rouazana) [inline mpt/impl/imap-mailbox/cyrus/pom.xml]: missing test scope?

File: mpt/impl/smtp/cassandra-rabbitmq-object-storage/pom.xml [hunk 2]
```diff
@@ -99,6 +113,10 @@
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.google.inject</groupId>
+            <artifactId>guice</artifactId>
```
Comment (rouazana) [inline mpt/impl/smtp/cassandra-rabbitmq-object-storage/pom.xml]: missing test scope?

File: mpt/impl/imap-mailbox/cyrus/pom.xml (hunk 1 above)
Comment (chibenwa) [reply mpt/impl/imap-mailbox/cyrus/pom.xml]: In a test project nobody depends on?

File: mpt/impl/smtp/cassandra-rabbitmq-object-storage/pom.xml (hunk 2 above)
Comment (chibenwa) [reply mpt/impl/smtp/cassandra-rabbitmq-object-storage/pom.xml]: In a test project nobody depends on?

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**